tests/functional/object/mcg/test_sts_client.py

```diff
@@ -0,0 +1,258 @@
+            * Try to assume the role
+
+        """
+        # change sts token expire time to 10 minute
+        add_env_vars_to_noobaa_core_class(
+            [(constants.STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS, 600000)]
+        )
+        logger.info("Changed the sts token expiration time to 10 minutes")
+
+        add_env_vars_to_noobaa_endpoint_class(
+            [(constants.STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS, 600000)]
+        )
+        logger.info("Changed the sts token expiration time to 10 minutes")
+
+        # create a bucket using noobaa admin creds
+        bucket_1 = "first-bucket"
+        retry(CommandFailed, tries=5, delay=5)(new_bucket)(bucket_name=bucket_1)
+        logger.info(f"Created bucket {bucket_1}")
+
+        # create a noobaa account
+        user_1 = f"user-{uuid4().hex}"
+        user_email_1 = f"{user_1}@email"
+        nb_account_1 = nb_account_factory(
+            mcg=mcg_obj_session,
+            user_name=f"{user_1}",
+            user_email=f"{user_email_1}",
+        )
+        signed_request_creds = {
+            "region": mcg_obj_session.region,
+            "endpoint": mcg_obj_session.sts_internal_endpoint,
+            "access_key_id": nb_account_1.access_key_id,
+            "access_key": nb_account_1.access_key,
+            "ssl": False,
+        }
+
+        # create another noobaa account
+        user_2 = f"user-{uuid4().hex}"
+        user_email_2 = f"{user_2}@email"
+        nb_account_2 = nb_account_factory(
+            mcg=mcg_obj_session,
+            user_name=f"{user_2}",
+            user_email=f"{user_email_2}",
+            allow_bucket_creation=False,
+        )
+        nb_user_access_key_id = nb_account_2.access_key_id
+        logger.info(f"Created new user '{user_2}'")
+
+        # assign assume role policy to the user-1
+        role_name = "user-1-assume-role"
+        nb_assign_user_role_fixture(user_email_2, role_name, principal=user_email_1)
+        logger.info(f"Assigned the assume role policy to the user {user_2}")
+
+        # noobaa admin assumes the above role
+        creds_generated = sts_assume_role(
+            awscli_pod_session,
+            role_name,
+            nb_user_access_key_id,
+            signed_request_creds=signed_request_creds,
+        )
+        mcg_obj_session.create_s3client_assumed_role(creds_generated)
+        logger.info(creds_generated)
+
+        # perform io to validate the role assumption
+        bucket_2 = "second-bucket"
+        try:
+            new_bucket(bucket_2, mcg_obj_session.assumed_s3_client)
+            assert (
+                False
+            ), "Bucket was created even though assumed role user doesnt have ability to create new bucket"
+        except Exception as err:
+            if "AccessDenied" not in err.args[0]:
+                raise
+            logger.info("Bucket creation failed as expected")
+        logger.info(s3_list_buckets(mcg_obj_session, mcg_obj_session.assumed_s3_client))
```

Comment:
Is this log a leftover from development? If not, please add content to the log that would clarify why we're listing the buckets when new no buckets were supposed to get created